vertexai/genai/tokenizer/corpora_test.go

```diff
@@ -241,47 +242,66 @@ func TestCountTokensWithCorpora(t *testing.T) {
 	model := client.GenerativeModel(defaultModel)
 	ucr := newUdhrCorpus()
 
+	tok, err := New(defaultModel)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	corporaURL := "https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/packages/corpora/udhr.zip"
-	files, err := corporaGenerator(corporaURL)
+	corporaFiles, err := corporaGenerator(corporaURL)
 	if err != nil {
 		t.Fatalf("Failed to generate corpora: %v", err)
 	}
 
-	// Iterate over files generated by the generator function
-	for _, fileInfo := range files {
-		if ucr.shouldSkip(fileInfo.Name) {
-			fmt.Printf("Skipping file: %s\n", fileInfo.Name)
-			continue
-		}
+	// Create channels for work distribution and results collection
+	corporaChan := make(chan corporaInfo)
+	doneChan := make(chan bool)
 
-		enc, found := ucr.getEncoding(fileInfo.Name)
-		if !found {
-			fmt.Printf("No encoding found for file: %s\n", fileInfo.Name)
-			continue
-		}
+	worker := func() {
+		for corpora := range corporaChan {
+			if ucr.shouldSkip(corpora.Name) {
+				fmt.Printf("Skipping file: %s\n", corpora.Name)
```

Comment:
`fmt` is not usually concurrency-safe (nothing bad will happen but output can get mixed up between goroutines).

Use `log.Printf` inside goroutines instead

Reply:
Done